rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupportImpl.java

```diff
@@ -0,0 +1,209 @@
+package co.rsk.peg.whitelist;
+
+import static co.rsk.peg.whitelist.WhitelistResponseCode.ADDRESS_ALREADY_WHITELISTED;
+import static co.rsk.peg.whitelist.WhitelistResponseCode.ADDRESS_NOT_EXIST;
+import static co.rsk.peg.whitelist.WhitelistResponseCode.DELAY_ALREADY_SET;
+import static co.rsk.peg.whitelist.WhitelistResponseCode.DISABLE_BLOCK_DELAY_INVALID;
+import static co.rsk.peg.whitelist.WhitelistResponseCode.GENERIC_ERROR;
+import static co.rsk.peg.whitelist.WhitelistResponseCode.INVALID_ADDRESS_FORMAT;
+import static co.rsk.peg.whitelist.WhitelistResponseCode.SUCCESS;
+import static co.rsk.peg.whitelist.WhitelistResponseCode.UNKNOWN_ERROR;
+
+import co.rsk.bitcoinj.core.Address;
+import co.rsk.bitcoinj.core.AddressFormatException;
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.bitcoinj.core.Context;
+import co.rsk.bitcoinj.store.BlockStoreException;
+import co.rsk.panic.PanicProcessor;
+import co.rsk.peg.vote.AddressBasedAuthorizer;
+import co.rsk.peg.whitelist.constants.WhitelistConstants;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.List;
+import org.ethereum.core.SignatureCache;
+import org.ethereum.core.Transaction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of the WhitelistSupport interface. Contains all the whitelist business logic.
+ */
+public class WhitelistSupportImpl implements WhitelistSupport {
+
+    private static final Logger logger = LoggerFactory.getLogger(WhitelistSupportImpl.class);
+    private static final String INVALID_ADDRESS_FORMAT_MESSAGE = "invalid address format";
+    private static final PanicProcessor panicProcessor = new PanicProcessor();
+    private final SignatureCache signatureCache;
+    private final Context btcContext;
+    private final WhitelistConstants whitelistConstants;
+    private final WhitelistStorageProvider whitelistStorageProvider;
+
+    public WhitelistSupportImpl(WhitelistStorageProvider whitelistStorageProvider,
+        WhitelistConstants whitelistConstants,
+        SignatureCache signatureCache, Context btcContext) {
```

Comment:
```suggestion
    public WhitelistSupportImpl(
    WhitelistStorageProvider whitelistStorageProvider,
        WhitelistConstants whitelistConstants,
        SignatureCache signatureCache, 
        Context btcContext) {
```
Looks more consistent this way

Reply:
Done!